.github/workflows/api-swagger-ui.yml

```diff
@@ -0,0 +1,76 @@
+name: Build and Push Swagger-UI to R2 mainnet-api-docs.spacemesh.network
+
+on:
+  release:
+    types: [published]
+    
+jobs:
+  check-version:
+    runs-on: ubuntu-22.04
+    outputs:
+      go-sm-api-version: ${{ steps.go-sm-api-version.outputs.GO_SM_API_VERSION }}
+    steps:
+    - name: Checkout target repository on last release
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        ref: 'refs/tags/${{ github.event.release.tag_name }}'
+    
+    - name: Extract dependency version
+      id: go-sm-api-version
+      run: |
+        version=$(grep -E 'github.com/spacemeshos/api/release/go' go.mod | awk '{print $2}')
+        echo "GO_SM_API_VERSION=$version" > $GITHUB_OUTPUT
+
+  build-deploy:
+    runs-on: ubuntu-22.04
+    needs: check-version
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: spacemeshos/api
+        path: api
+        fetch-depth: 0
+        ref: 'refs/tags/${{ needs.check-version.outputs.go-sm-api-version }}'
+    - name: Use Node.js v20.14.0
+      uses: actions/setup-node@v4
+      with:
+        node-version: '20.14.0'
+
+    - name: npm install
+      run: npm install
+      working-directory: api/release/openapi/swagger
+
+    - name: npm build
+      run: npm run build
+      working-directory: api/release/openapi/swagger
+
+    - uses: actions/upload-artifact@v4
+      with:
+        name: swagger-ui-build
+        path: api/release/openapi/swagger/dist
+
+    - name: upload to mainnet
+      uses: jakejarvis/s3-sync-action@master
+      with:
+        args: --acl public-read --follow-symlinks --delete
+      env:
+          AWS_S3_BUCKET: ${{ secrets.CLOUDFLARE_API_BUCKET }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.CLOUDFLARE_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.CLOUDFLARE_SECRET_ACCESS_KEY }}
+          SOURCE_DIR: api/release/openapi/swagger/dist
+          DEST_DIR: ''
+          AWS_S3_ENDPOINT: https://${{ secrets.CLOUDFLARE_ACCOUNT_ID }}.r2.cloudflarestorage.com
+
+    - name: upload to testnet
+      uses: jakejarvis/s3-sync-action@master
+      with:
+        args: --acl public-read --follow-symlinks --delete
+      env:
+          AWS_S3_BUCKET: ${{ secrets.CLOUDFLARE_TESTNET_API_BUCKET }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.CLOUDFLARE_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.CLOUDFLARE_SECRET_ACCESS_KEY }}
+          SOURCE_DIR: api/release/openapi/swagger/dist
+          DEST_DIR: ''
+          AWS_S3_ENDPOINT: https://${{ secrets.CLOUDFLARE_ACCOUNT_ID }}.r2.cloudflarestorage.com
```

Comment:
We definitely don't use the same versions on mainnet and testnet. Can this be changed that new versions don't overwrite the existing api definition?

Reply:
Currently the host on api swagger spec is only targeting testnet-api, so for now, all releases are getting uploaded to testnet-api-docs.spacemesh.network. The way it's going to be separated between mainnet and testnet needs to be talked about